.gitignore

```diff
@@ -1,4 +1,54 @@
-.idea
-.kitchen
 .DS_Store
 build/
+
+*.rbc
```

Comment:
Are these all required? I'd rather keep .gitignore lean to expose when new files are being brought into a workspace which can highlight their ramifications to the project.

Reply:
Most of 'em are added by default when running `chef generate cookbook` , some others are future-proofing. I could easily pare it down to match Chef defaults.

Reply:
This reminds me of the discussion here [0]. I would prefer only adding entries in .gitignore for files that are created during the development process that should not be committed to source control. This may mean omitting some that chef creates by default.

[0] https://github.com/sonatype/chef-nexus-repository-manager/pull/1#discussion_r137826127

Reply:
Even if those defaults are prescribed, based on extensive development, by the upstream creators? ex: Chef? Some of the ignore entries are also future-proofing.

Reply:
Or they have an opinionated process which is defensive against letting non source files into GitHub whereas we have one which is defensive against misunderstood intent. /me shrugs.